impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -458,6 +458,170 @@ define(["./_focus"], function(focusUtil) {
 
   }
 
+  function makeAccessibleToolbar(elem) {
+    elem.setAttribute("role", "toolbar");
+    elem.setAttribute("tabindex", "0");
+
+    var firstItem = null;
+    var lastItem = null;
+
+    var toolbarItems = [];
+
+    elem.addEventListener("keydown", onKeyPress);
+    elem.addEventListener("focus", onContainerFocus);
+
+    items = elem.querySelectorAll('.toolbarButton');
+
+    for (i = 0; i < items.length; i++) {
+      var toolbarItem = items[i];
+
+      if (firstItem == null && focusUtil.isTabbable(toolbarItem) && !isDisabled(toolbarItem)) {
+        firstItem = toolbarItem;
+        firstItem.setAttribute('tabindex', '0');
+      } else {
+        toolbarItem.setAttribute('tabindex', '-1');
+      }
+      lastItem = toolbarItem;
+      toolbarItems.push(toolbarItem);
+    }
+
+    return {
+      destroy: destroy
+    };
+
+    function destroy() {
+      elem.removeEventListener("keydown", onKeyPress);
+    }
+
+    function onContainerFocus() {
+      elem.setAttribute("tabindex", "-1");
+      setFocusToFirst();
+    }
+
+    function isDisabled(item) {
+      var items = Array.from(item.classList);
+      // for cases when class is 'disabled' or 'pentaho-imagebutton-disabled'
+      return items.some(function (child) {
+        return child.includes('disabled');
+      });
+    }
+
+    function isHidden(item) {
+      return item.classList.contains('hidden');
+    }
+
+    function isNotFocusable(item) {
+      return isDisabled(item) || isHidden(item);
+    }
+
+    function setFocusItem(item) {
+      for (var i = 0; i < toolbarItems.length; i++) {
+        toolbarItems[i].setAttribute('tabindex', '-1');
+      }
+
+      if (isDisabled(item)) { //for the case when firstitem is disabled
+        item = getNext(item);
+      }
+
+      item.setAttribute('tabindex', '0');
+      item.focus();
+    }
+
+    function getNext(currentItem) {
+      var index, newItem, length = toolbarItems.length;
+
+      index = toolbarItems.indexOf(currentItem);
+      if (index < 0 || index === (length - 1)) {
+        newItem = toolbarItems[0];
+      } else {
+        newItem = toolbarItems[index + 1];
+      }
+
+      if (isNotFocusable(newItem)) {
+        return getNext(newItem);
+      }
+      return newItem;
+    }
+
+    function getPrevious(currentItem) {
+      var index, newItem, length = toolbarItems.length;
+
+      index = toolbarItems.indexOf(currentItem);
+      if (index <= 0) {
+        newItem = toolbarItems[length - 1];
+      } else {
+        newItem = toolbarItems[index - 1];
+      }
+
+      if (isNotFocusable(newItem)) {
+        return getPrevious(newItem);
+      }
+      return newItem;
+    }
+
+    function setFocusToFirst() {
+      setFocusItem(firstItem);
```

Comment:
The first item should be determined dynamically, each time. Don't think the performance benefit offered by the proposed solution is worth the added complexity.